syncstorage/storage/sql/__init__.py

```diff
@@ -766,6 +769,16 @@ def _purge_expired_batches(self, grace_period=0, max_per_loop=1000):
             "grace": grace_period,
             "maxitems": max_per_loop,
         })
+        # For TokuDB tables, doing a bunch of in-order deletes can cause
+        # performance degradation of the indexes, as described in:
+        #
+        #   https://www.percona.com/blog/2015/02/11/tokudb-table-optimization-improvements/
+        #
+        # We run an `OPTIMIZE TABLE` after each purge to keep this in check.
+        if self._optimize_table_after_purge:
+            if self.dbconnector.driver == "mysql":
```

Comment:
What do you think about moving these conditions into a Boolean getter on `SQLStorage`, so that you don't need to duplicate the comment?